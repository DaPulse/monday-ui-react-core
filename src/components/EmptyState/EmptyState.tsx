import React, { forwardRef, useRef } from "react";
import cx from "classnames";
import { useMergeRefs } from "../../hooks";
import VibeComponentProps from "../../types/VibeComponentProps";
import VibeComponent from "../../types/VibeComponent";
import { getTestId } from "../../tests/test-ids-utils";
import { ComponentDefaultTestId } from "../../tests/constants";
import styles from "./EmptyState.module.scss";
import Flex from "../Flex/Flex";
import Heading from "../Heading/Heading";
import Text from "../Text/Text";
import Button from "../Button/Button";

export interface EmptyStateProps extends VibeComponentProps {
  imgSrc: string;
  title: string;
  body: string;
  onPrimaryActionClick?: () => void;
  primaryActionLabel?: string;
  onSecondaryActionClick?: () => void;
  secondaryActionLabel?: string;
  imgClassName?: string;
}

const EmptyState: VibeComponent<EmptyStateProps, HTMLElement> = forwardRef(
  (
    {
      imgSrc,
      title,
      body,
      onPrimaryActionClick,
      primaryActionLabel,
      onSecondaryActionClick,
      secondaryActionLabel,
      className,
      imgClassName,
      id,
      "data-testid": dataTestId
    },
    ref
  ) => {
    const componentRef = useRef(null);
    const mergedRef = useMergeRefs({ refs: [ref, componentRef] });

    return (
      <Flex
        ref={mergedRef}
        className={cx(styles.emptyState, className)}
        direction={Flex.directions.COLUMN}
        align={Flex.align.CENTER}
        justify={Flex.justify.CENTER}
        gap={Flex.gaps.LARGE}
        id={id}
        data-testid={dataTestId || getTestId(ComponentDefaultTestId.EMPTY_STATE, id)}
      >
        <img src={imgSrc} alt={title} className={cx(styles.image, imgClassName)} />
        <Flex direction={Flex.directions.COLUMN} gap={Flex.gaps.SMALL} align={Flex.align.CENTER}>
          <Heading type={Heading.types.H2}>{title}</Heading>
          <Text type={Text.types.TEXT1}>{body}</Text>
        </Flex>
        <Flex gap={Flex.gaps.SMALL} align={Flex.align.CENTER}>
          {secondaryActionLabel && (
            <Button kind={Button.kinds.TERTIARY} onClick={onSecondaryActionClick}>
              {secondaryActionLabel}
            </Button>
          )}
          {primaryActionLabel && <Button onClick={onPrimaryActionClick}>{primaryActionLabel}</Button>}
        </Flex>
      </Flex>
    );
  }
);

export default EmptyState;
