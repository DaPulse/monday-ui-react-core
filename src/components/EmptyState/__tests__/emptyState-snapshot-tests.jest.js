import React from "react";
import renderer from "react-test-renderer";
import EmptyState from "../EmptyState";

const defaultProps = {
  imgSrc: "someImg",
  title: "This is title",
  body: "This is body"
};

describe("EmptyState", () => {
  it("should render correctly with required props", () => {
    const tree = renderer.create(<EmptyState {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
