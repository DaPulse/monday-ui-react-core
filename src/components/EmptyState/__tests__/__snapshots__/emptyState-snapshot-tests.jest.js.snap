// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState should render correctly with required props 1`] = `
<div
  className="container directionColumn justifyCenter alignCenter emptyState"
  data-testid="empty-state"
  style={
    Object {
      "gap": "24px",
    }
  }
>
  <img
    alt="This is title"
    className="image"
    src="someImg"
  />
  <div
    className="container directionColumn justifyStart alignCenter"
    style={
      Object {
        "gap": "8px",
      }
    }
  >
    <h2
      className="typography primary start singleLineEllipsis heading h2Normal"
      data-testid="text"
    >
      This is title
    </h2>
    <div
      className="typography primary start singleLineEllipsis text text1Normal"
      data-testid="text"
    >
      This is body
    </div>
  </div>
  <div
    className="container directionRow justifyStart alignCenter"
    style={
      Object {
        "gap": "8px",
      }
    }
  />
</div>
`;
